toast-tk-rest-utils/src/main/java/io/toast/tk/core/rest/RestUtils.java

```diff
@@ -158,49 +177,83 @@ public static ImportedScenarioDescriptor getScenario(final ImportedScenario scen
+			httppost.setHeader("Token" , ApiKey);
 			
-			final WebResource webResource = httpClient.resource(url);
-			final ClientResponse response = webResource.type(MediaType.APPLICATION_JSON)
-														.accept(MediaType.APPLICATION_JSON)
-														.post(ClientResponse.class, record);
-			return response.getStatus() == 200;
+			StringEntity  input = new StringEntity (jsonFixtureDescriptor);
+		    httppost.setEntity(input);
+			response = httpClient.execute(httppost);
+			if (response.getStatusLine().getStatusCode() != 200) {
+				throw new IllegalStateException("Failed to post connectors - HTTP reply code : "  + response.getStatusLine().getStatusCode());
+			}	
+			return response.getStatusLine().getStatusCode() == 200;
 		} catch (final Exception e) {
 			LOG.error(e.getMessage(), e);
-			return false;
-		}
+		} 
+		return false;
 	}
-
-	public static void main(final String[] args) {
-		RestUtils.postScenario("newtest", "localhost", "9000", "a step");
+	
+	public static boolean postWebEventRecord(final String url, final String record, final String ApiKey) {	
+		try(final CloseableHttpClient httpClient = HttpClients.createDefault()) {
+			if(Strings.isNullOrEmpty(ApiKey)) {
+				throw new IllegalAccessException("No Api Key provided !");
+			}
+			HttpPost httppost = new HttpPost(url);			
+			httppost.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+			httppost.setHeader("Token" , ApiKey);
+			
+			StringEntity  input = new StringEntity (record);
+		    httppost.setEntity(input);
+			response = httpClient.execute(httppost);
+			if (response.getStatusLine().getStatusCode() != 200) {
+				throw new IllegalStateException("Failed to publish web record - HTTP reply code : "  + response.getStatusLine().getStatusCode());
+			}	
+			return response.getStatusLine().getStatusCode() == 200;
+		} catch (final Exception e) {
+			LOG.error(e.getMessage(), e);
+		} 
+		return false;
 	}
+	
+	public static boolean registerAgent(final String url, String information, final String ApiKey) {
+		try(final CloseableHttpClient httpClient = HttpClients.createDefault()) {
+			if(Strings.isNullOrEmpty(ApiKey)) {
+				throw new IllegalAccessException("No Api Key provided !");
+			}
+			
+			HttpPost httppost = new HttpPost(url);	
+			httppost.setHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON);
+			httppost.setHeader("Token" , ApiKey);
 
-	public static boolean registerAgent(final String url, String information) {
-		try {
-			final Client httpClient = Client.create();
-			final WebResource webResource = httpClient.resource(url);
-			final ClientResponse response = webResource.type(MediaType.APPLICATION_JSON)
-														.accept(MediaType.APPLICATION_JSON)
-														.post(ClientResponse.class, information);
-			return response.getStatus() == 200;
+			StringEntity input = new StringEntity(information);
+			input.setContentType("application/json");
+			httppost.setEntity(input);
+			
+			response = httpClient.execute(httppost);

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Synchronize this lazy initialization of 'response' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.talanlabs.com/coding_rules#rule_key=squid%3AS2444)